certificate_loader_test.go

```diff
@@ -81,7 +81,104 @@ func testCertificateLoader(t *testing.T, context spec.G, it spec.S) {
 		})
 	})
 
-	context("load", func() {
+	context("load pkcs12", func() {
+		var (
+			path string
+		)
+
+		it.Before(func() {
+			in, err := os.Open(filepath.Join("testdata", "test-keystore.pkcs12"))
+			Expect(err).NotTo(HaveOccurred())
+			defer in.Close()
+
+			out, err := os.CreateTemp("", "certificate-loader")
+			Expect(err).NotTo(HaveOccurred())
+			defer out.Close()
+
+			_, err = io.Copy(out, in)
+			Expect(err).NotTo(HaveOccurred())
+
+			path = out.Name()
+		})
+
+		it.After(func() {
+			Expect(os.RemoveAll(path)).To(Succeed())
+		})
+
+		it("ignores non-existent file", func() {
+			c := libjvm.CertificateLoader{
+				CertFile: filepath.Join("testdata", "non-existent-file"),
+				Logger:   io.Discard,
+			}
+
+			Expect(c.Load(path, "changeit")).To(Succeed())
+		})
+
+		it("ignores non-existent directory", func() {
+			c := libjvm.CertificateLoader{
+				CertDirs: []string{filepath.Join("testdata", "non-existent-directory")},
+				Logger:   io.Discard,
+			}
+
+			Expect(c.Load(path, "changeit")).To(Succeed())
+		})
+
+		it("loads additional certificates from file", func() {
+			c := libjvm.CertificateLoader{
+				CertFile: filepath.Join("testdata", "certificates", "certificate-1.pem"),
+				Logger:   io.Discard,
+			}
+
+			Expect(c.Load(path, "changeit")).To(Succeed())
+
+			in, err := os.ReadFile(path)
+			Expect(err).NotTo(HaveOccurred())
+
+			ks, err := pkcs12.DecodeTrustStore(in, "")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ks).To(HaveLen(2))
+		})
+
+		it("loads additional certificates from directories", func() {
+			c := libjvm.CertificateLoader{
+				CertDirs: []string{filepath.Join("testdata", "certificates")},
+				Logger:   io.Discard,
+			}
+
+			Expect(c.Load(path, "changeit")).To(Succeed())
+
+			in, err := os.ReadFile(path)
+			Expect(err).NotTo(HaveOccurred())
+
+			ks, err := pkcs12.DecodeTrustStore(in, "")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(ks).To(HaveLen(3))
+		})
+
+		if internal.IsRoot() {
+			return
+		}
```

Comment:
I know we do this in some other places, but I'm not a fan of this.

My understanding is that it's a way to stop running tests when the tests are running as root. The idea being that would give false positives because root can write to anything.

I think this is very unintuitive though and it's also not clear when/if that test skips. It could be skipping every time we run, and we'd never know.

Could move this into the actual test and [call `t.Skip()` instead](https://stackoverflow.com/a/50631395/1585136)? That would give us better output, I think.

Reply:
I guess `t.Skip` will not work with the `github.com/sclevine/spec` framework (it would skip the whole suite, wouldn't it?)  
Anyway, the framework has its own way of making tests as "pending", which we use now.

Reply:
Thanks, that change is fantastic!